runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -542,13 +543,16 @@ setCallSiteTargetImpl(J9VMThread *currentThread, jobject callsite, jobject targe
 		MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
 
 		/* Check for MutableCallSite modification. */
-		J9JITConfig* jitConfig = currentThread->javaVM->jitConfig;
-		J9UTF8 *clazzNameUTF8 = J9ROMCLASS_CLASSNAME(clazz->romClass);
-		U_8 *clazzName = J9UTF8_DATA(clazzNameUTF8);
-		UDATA clazzLength = J9UTF8_LENGTH(clazzNameUTF8);
+		J9JITConfig* jitConfig = javaVM->jitConfig;
+		J9Class *mcsClass = vmFuncs->hashClassTableAt(
```

Comment:
This isn't safe though as the current thread must be holding the `vm->classTableMutex` to be able to examine the `hashClassTableAt`.  Use `peekClassHashTable` instead to handle getting and releasing the mutex (and avoiding it in the special cases where it's safe to)